dsms/knowledge/utils.py

```diff
@@ -2,48 +2,162 @@
 import io
 import json
 import re
+import warnings
 from enum import Enum
 from pathlib import Path
 from typing import TYPE_CHECKING, Any, Dict, List, Optional, Union
 from uuid import UUID
 
 import pandas as pd
-from pydantic import BaseModel, create_model
 from requests import Response
 
-from dsms.core.utils import _name_to_camel, _perform_request
+from pydantic import (  # isort: skip
+    BaseModel,
+    ConfigDict,
+    Field,
+    create_model,
+    model_validator,
+)
+
+from dsms.core.utils import _name_to_camel, _perform_request  # isort:skip
+
+from dsms.knowledge.properties.custom_datatype import (  # isort:skip
+    NumericalDataType,
+)
 
 if TYPE_CHECKING:
     from dsms.core.context import Buffers
     from dsms.knowledge import KItem, KType
 
 
-def _parse_model(value: Dict[str, Any]) -> BaseModel:
+def _is_number(value):
+    try:
+        float(value)
+        return True
+    except Exception:
+        return False
+
+
+def _create_custom_properties_model(
+    value: Optional[Dict[str, Any]]
+) -> BaseModel:
     """Convert the dict with the model schema into a pydantic model."""
-    title = value.get("title")
-    properties = value.get("properties")
-    if not title:
-        raise KeyError("`data_schema` does not have any `title`.")
-    if isinstance(properties, type(None)):
-        raise KeyError("`data_schema` does not have any `properties`.")
+    from dsms import KItem
+
     fields = {}
-    for key, props in properties.items():
-        dtype = props.get("type")
-        default = props.get("default")
-        if dtype == "string":
-            dtype = str
-        elif dtype == "integer":
-            dtype = int
-        elif dtype == "float":
-            dtype = float
-        elif dtype == "bool":
-            dtype = bool
-        elif dtype == "object":
-            dtype = dict
-        else:
-            raise TypeError(f"Invalid `type={dtype}`")
-        fields[key] = (dtype, ... if not default else default)
-    return create_model(title, **fields)
+    if isinstance(value, dict):
```

Comment:
I had to comment out this block due to the latest changes done related to the integration of form2rdf into DSMS